### ISSUE TITLE
Implement a better loss function

### ISSUE THREAD

**Issue by josura** [josura/GNN-metapathways#1]
Right now I have some problems with the better loss function that I was trying to implement, my laptop GPU is not enough.

**Comment by josura** [comment]
Right now there are some problems with the new and better loss function that considers all pairs, that is

<img src="https://render.githubusercontent.com/render/math?math={\sum_{x\in V}\sum_{y\in V}c ||z_x - z_y||_2}#gh-light-mode-only">
<img src="https://render.githubusercontent.com/render/math?math={\sum_{x\in V}\sum_{y\in V}c ||z_x - z_y||_2}#gh-dark-mode-only">


All pairs of genes are a lot. the division in batches also does not seem to work well but the fault is of the implementation of the whole process of computing the difference of genes and its implementation in tensorflow.

**Comment by josura** [comment]
I implemented a better loss function for gene and pathway embeddings.
